### PR TITLE
Add myself as a dep reviewer

### PR DESCRIPTION
**Pull request by logicalhan**
```release-note
NONE
```

### CODE REVIEW

File: vendor/OWNERS [hunk 1]
```diff
@@ -2,3 +2,5 @@
 
 approvers:
 - dep-approvers
+reviewers:
+- logicalhan
```
Comment (liggitt) [inline vendor/OWNERS:6]: I'm a big +1 on expanding the reviewer pool. 

That said, this file is generated, and we probably want to manage this as a group.

I'd suggest adding a `dep-reviewers` alias, and updating `update-vendor.sh` to add
```
reviewers:
- dep-reviewers
```

**Comment by logicalhan** [comment]
/kind cleanup

I have no idea which sig to tag this to..
/priority backlog

**Comment by logicalhan** [comment]
/sig api-machinery
(there is no sig for deps) 
/triage accepted

**Comment by dims** [comment]
@logicalhan can you please make it to the sig-arch code-organization meetings?

**Comment by logicalhan** [comment]
> @logicalhan can you please make it to the sig-arch code-organization meetings? 

Sure, I'll make it a point to attend those :)

**Pull request merged**